File: android/app/src/main/java/com/djs66256/short_drama/data/dto/ToggleCommentLikeResponseDto.kt
package com.djs66256.short_drama.data.dto

import com.djs66256.short_drama.domain.model.ToggleCommentLikeResult
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ToggleCommentLikeResponseDto(
    @SerialName("comment_id")
    val commentId: String,
    val liked: Boolean,
    @SerialName("like_count")
    val likeCount: Int,
) {
    fun toDomain(): ToggleCommentLikeResult {
        return ToggleCommentLikeResult(
            commentId = commentId,
            liked = liked,
            likeCount = likeCount,
        )
    }
}

File: ios/ShortDrama/Sources/Data/DTOs/ToggleCommentLikeResponseDTO.swift
import Foundation

struct ToggleCommentLikeResponseDTO: Codable, Equatable {
    let commentId: String
    let liked: Bool
    let likeCount: Int

    func toEntity() -> ToggleCommentLikeResult {
        ToggleCommentLikeResult(commentId: commentId, liked: liked, likeCount: likeCount)
    }
}
